apps/viptt/models/service-agreements.js

```diff
@@ -24,15 +22,91 @@ module.exports = class ServiceAgreements {
     this._serviceAgreementData = slaData[slaVersion];
   }
 
+  validateDatasetLoaded(data) {
+    if (!data) {
+      throw new Error('SLA Dataset not loaded');
+    }
+  }
+
+  validateStringArg(value, name) {
+    if (typeof value !== 'string') {
+      throw new Error(`Invalid arguments: ${name} must be a string`);
+    }
+  }
+
+  validatePositiveNumeric(value, fieldName, bucketKey, optionKey) {
+    const raw = typeof value === 'string' ? value.trim() : value;
+    if (raw === null || raw === '') {
+      throw new Error(`Invalid SLA entry for ${bucketKey}.${optionKey}: missing ${fieldName}`);
+    }
+    const num = Number(raw);
+    if (!Number.isFinite(num) || num <= 0) {
+      throw new Error(`Invalid SLA entry for ${bucketKey}.${optionKey}: invalid ${fieldName} '${value}'`);
+    }
+  }
+
+  validateSLAFields(option, bucketKey, optionKey) {
+    this.validatePositiveNumeric(option.days, 'days', bucketKey, optionKey);
+    this.validatePositiveNumeric(option.weeks, 'weeks', bucketKey, optionKey);
+  }
+
   /**
-   * Return the SLA data according to the user's reason for applying
-   * and whether they are inside or outside the UK
-   * @param {string} reason
-   * @param {boolean} insideUK
-   * @returns {string | null}
+   * Retrieve a top-level SLA bucket or throw if missing/invalid.
+   *
+   * @param {Object} data - The full SLA dataset.
+   * @param {string} bucketKey - Bucket key, e.g. `'inside-uk'`, `'outside-uk'`, `'premium'`.
+   * @returns {Object} The bucket object containing SLA options.
+   * @throws {Error} If the bucket is missing or not an object.
    */
-  getSLAData(reason, insideUK = true) {
-    const ukStatus = insideUK ? INSIDE_UK : OUTSIDE_UK;
-    return this._serviceAgreementData[ukStatus][reason] || null;
+  getSLABucket(data, bucketKey) {
```

Comment:
As mentioned above could do with a clearer name than 'bucket' and 'bucketKey' for the different types of applications